### DISCUSSION TITLE
Is it possible to not have a success redirect upon authenticate?

### DISCUSSION THREAD

**Discussion by 684efs3** [sergiodxa/remix-auth#254] [Q&A] [answered]
I am using the Remix Email Strategy, and I am trying to get it to authenticate (send a magic link) without the need to redirect upon success. On digging in, I think this requirement is in the remix-auth package (?). Is there a way to no redirect? I am using a dropdown login, and I don't want the change the page when a magic link has been sent. Thank you!

**Comment by dvnrsn** [comment, marked as answer]
Hey @684efs3, successRedirect is not required. You can see an example [here](https://github.com/sergiodxa/remix-auth#custom-redirect-url-based-on-the-user). Calling `authenticate` will simply return the user (or whatever your strategy function returns).

**Comment by 684efs3** [comment]
This is my code for login.tsx

```ts
// Libraries
import { json } from "@remix-run/node";
import { auth } from "~/services/auth.server";
import { sessionStorage } from "~/services/session.server";

export let loader = async ({ request }) => {
  await auth.isAuthenticated(request, { successRedirect: "/profile" });
  let session = await sessionStorage.getSession(request.headers.get("Cookie"));
  return json({
    magicLinkSent: session.has("auth:magiclink"),
    magicLinkEmail: session.get("auth:email"),
  });
};

export let action = async ({ request }) => {
  await auth.authenticate("email-link", request, {
    // successRedirect: "/",
  });
};
```

If I remove `successRedirect` I get this error in my browser:

```
Error: Missing successRedirect. The successRedirect is required for POST requests.
```

Is this built into Remix Auth, or is this something that [Remix Auth Email Strategy](https://github.com/pbteja1998/remix-auth-email-link) built in? Their docs say:

```ts
export let action = async ({ request }: ActionArgs) => {
  // The success redirect is required in this action, this is where the user is
  // going to be redirected after the magic link is sent, note that here the
  // user is not yet authenticated, so you can't send it to a private page.
  await auth.authenticate('email-link', request, {
    successRedirect: '/login',
    // If this is not set, any error will be throw and the ErrorBoundary will be
    // rendered.
    failureRedirect: '/login',
  })
}
```

**Reply by dvnrsn** [reply]
Consider using ```ts
to syntax highlight for easier reading
```ts
  export let action = async ({ request }: ActionArgs) => {

**Comment by dvnrsn** [comment]
https://github.com/pbteja1998/remix-auth-email-link/issues/45
